models/lwr/openmc_materials.py

```diff
@@ -0,0 +1,109 @@
+#--------------------------------------------------------------------------------------------------------------------------#
+# This is based on the C5G7 reactor physics benchmark problem extension case as described in:                              #
+# "Benchmark on Deterministic Transport Calculations Without Spatial Homogenisation: MOX Fuel Assembly 3-D Extension Case" #
+# [NEA/NSC/DOC(2005)16]                                                                                                    #
+# https://www.oecd-nea.org/upload/docs/application/pdf/2019-12/nsc-doc2005-16.pdf                                          #
+#                                                                                                                          #
+# The original C5G7 benchmark is defined with multi-group cross sections. To account for                                   #
+# continuous energy spectral effects, we chose to use the material properties provided in:                                 #
+# "Proposal for a Second Stage of the Benchmark on Power Distributions Within Assemblies"                                  #
+# [NEA/NSC/DOC(96)2]                                                                                                       #
+# https://www.oecd-nea.org/upload/docs/application/pdf/2020-01/nsc-doc96-02-rev2.pdf                                       #
+#--------------------------------------------------------------------------------------------------------------------------#
+
+import openmc
+import numpy as np
+
+#--------------------------------------------------------------------------------------------------------------------------#
+# Material definitions.
+## First fuel region: 4.3% MOX.
+mox_4_3_comp = 1.0e24 * np.array([5.00e-5, 2.21e-2, 1.50e-5, 5.80e-4, 2.40e-4, 9.80e-5, 5.40e-5, 1.30e-5, 4.63e-2])
+mox_4_3_frac = mox_4_3_comp / np.sum(mox_4_3_comp)
+mox_4_3 = openmc.Material(name = '4.3% MOX Fuel', temperature = 293.15)
+mox_4_3.add_nuclide('U235', mox_4_3_frac[0], percent_type = 'ao')
+mox_4_3.add_nuclide('U238', mox_4_3_frac[1], percent_type = 'ao')
+mox_4_3.add_nuclide('Pu238', mox_4_3_frac[2], percent_type = 'ao')
+mox_4_3.add_nuclide('Pu239', mox_4_3_frac[3], percent_type = 'ao')
+mox_4_3.add_nuclide('Pu240', mox_4_3_frac[4], percent_type = 'ao')
+mox_4_3.add_nuclide('Pu241', mox_4_3_frac[5], percent_type = 'ao')
+mox_4_3.add_nuclide('Pu242', mox_4_3_frac[6], percent_type = 'ao')
+mox_4_3.add_nuclide('Am241', mox_4_3_frac[7], percent_type = 'ao')
+mox_4_3.add_element('O', mox_4_3_frac[8], percent_type = 'ao')
+mox_4_3.set_density('atom/cm3', np.sum(mox_4_3_comp))
```

Comment:
I worry about this being error prone since you rely on the indexing of the comp vector to line up with the order they are used.  Another alternative could be something like this (subject to final debugging):
```suggestion
mox_4_3_comp = {
  'U235': 5.00e-5,
  'U238': 2.21e-2,
  'Pu238': 1.50e-5,
  'Pu239': 5.80e-4,
  'Pu240': 2.40e-4,
  'Pu241': 9.80e-5,
  'Pu242': 5.40e-5,
  'Am241': 1.30e-5,
  'O': 4.63e-2
  }
mox_4_3_density = 1.0e24 * np.array(mox_4_3_comp.values())
mox_4_3 = openmc.Material(name = '4.3% MOX Fuel', temperature = 293.15)
for comp, atom_frac in mox_4_3_comp.items():
    if comp[-1].isdigit():
        mox_4_3.add_nuclide(comp, atom_frac, percent_type = 'ao')
    else:
        mox_4_3.add_element(comp, atom_frac, percent_type = 'ao')
mox_4_3.set_density('atom/cm3', mox_4_3_density)
```

This would also allow you to reuse the composition dictionary as a starting point to define the other MOX vectors.

The next step might be to define the composition dictionaries and the material names in a data structure that just lets you loop over them all to generate them.

Reply:
I've gone ahead and implemented this approach alongside a loop over all compositions to populate a dictionary of OpenMC materials.